app/controllers/concerns/saml_idp_logout_concern.rb

```diff
@@ -48,7 +48,12 @@ def name_id_user
   end
 
   def sp_slo_identity
-    @_sp_slo_identity ||= Identity.includes(:user).find_by(uuid: name_id)
+    identity = if FeatureManagement.enable_agency_based_uuids?
```

Comment:
Memoizing here doesn't actually do anything since you still end up computing the value every time you call this. You could wrap this in a block to get around that:

```ruby
@_sp_slo_identity ||= begin
  if FeaturesManagement.things_enabled
    blah(blah)
  else
    blah :blah
  end
end
```

Reply:
Nice catch!  Originally I had it on one line and lint complained.  When I broke it up I inadvertently broke the memoization.